python/ray/train/batch_predictor.py

```diff
@@ -222,13 +281,19 @@ def __call__(self, batch):
                 # Set the in-predictor preprocessing to a no-op when using a separate
                 # GPU stage. Otherwise, the preprocessing will be applied twice.
                 override_prep = BatchMapper(lambda x: x)
+                # preprocessor.transform will break for DatasetPipeline due to
+                # missing _dataset_format()
```

Comment:
This is no longer true with your addition, we should try unifying these paths in a follow-up PR.